### PR TITLE
Moved useResizeObserver to utils

### LINKED ISSUE #863: Extract useResizeObserver hook into reusable util
# 🙋 Feature Request

As of #793 we have a nice `useResizeObserver` hook that is used in breadcrumbs to handle setting up a `ResizeObserver` where available with a basic fallback to window resize events. It would be great to move this hook into `@react-spectrum/utils` so we can reuse it in other components, e.g. ButtonGroup and eventually ActionGroup and Tabs.

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by McZenith**
Closes #863

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/863).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by devongovett** [comment]
Good start! I think you're going to need to move the ResizeObserver.d.ts file into the utils package as well to fix the TypeScript errors.

**Comment by McZenith** [comment]
@devongovett, thanks! I still get the same error after moving type declaration file though. is there a better work around this?

File: packages/@react-spectrum/utils/src/useResizeObserver.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (!ref) { return; }
+
+    if (!hasResizeObserver()) {
+      window.addEventListener('resize', onResize, false);
+      return () => {
+        window.removeEventListener('resize', onResize, false);
+      };
+    } else {
+
+      const resizeObserverInstance = new ResizeObserver((entries) => {
```
Comment (snowystinger) [inline packages/@react-spectrum/utils/src/useResizeObserver.ts:25]: i think this should be window.ResizeObserver, or both should be without. But it's strange for it to be two ways

**Comment by McZenith** [comment]
Thank you @snowystinger I will look into it.

**Comment by McZenith** [comment]
I think i found a way around it! so, I had to cast ResizeObserver to (<any>window) and it worked. but some tests from other components were making the CircleCI tests to fail, they, of course, are independent of the ResizeObserver!

**Comment by snowystinger** [comment]
O, some of those dependencies really are meant to be left out, we should really eslint disable those. It looks like you ran eslint fix?
If you can undo those in files you weren't working on, that should help the tests pass again

**Comment by McZenith** [comment]
Oh, yeah, that's true... I will make necessary changes and resend a PR
...

Thank you for your time...  I really appreciate!

**Comment by McZenith** [comment]
Done!!! @snowystinger, thanks for the help thus far. I really appreciate your kindness and patient!

**Comment by McZenith** [comment]
Thank you @snowystinger, I'm really glad it's my first time...

**Pull request merged**